File: npb_similar_player/analyze.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-

HOST = "127.0.0.1"
PORT = 9199
NAME = "recommender_baseball"
require 'jubatus/recommender/client'

recommender =
  Jubatus::Recommender::Client::Recommender.new(HOST, PORT)
File.open("dat/baseball.csv") {|f|
  f.each{|line|
    pname, team, bave, games, pa, atbat, hit, homerun, runsbat,
      stolen, bob, hbp, strikeout, sacrifice, dp, slg, obp,
      ops, rc27, xr27 = line.split(",")
    sr = recommender.similar_row_from_id(NAME, pname, 4)
    puts("player #{pname} is similar to : #{sr[1][0]} #{sr[2][0]} #{sr[3][0]}")
  }
}

File: npb_similar_player/python/analyze.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# Python 2 compatibility:
from __future__ import unicode_literals

import sys
import codecs
from jubatus.recommender import client
from jubatus.recommender import types

NAME = "recommender_baseball";

if __name__ == '__main__':

    recommender = client.Recommender("127.0.0.1", 9199, NAME)

    for line in codecs.open('../dat/baseball.csv', 'r', 'utf-8'):
      pname, team, bave, games, pa, atbat, hit, homerun, runsbat, stolen, bob, hbp, strikeout, sacrifice, dp, slg, obp, ops, rc27, xr27 = line[:-1].split(',')
      sr = recommender.similar_row_from_id(pname , 4)
      print("player {0} is similar to : {0}, {1}, {2}".format(pname, sr[1].id, sr[2].id, sr[3].id))

